File: Code/new_fitervari_backend/src/main/java/at/htl/control/CustomerRepository.java
package at.htl.control;

import at.htl.model.Customer;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class CustomerRepository implements PanacheRepository<Customer> {

    public Customer save(Customer customer) {
        return getEntityManager().merge(customer);
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/CustomerRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.Customer
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class CustomerRepository: PanacheRepository<Customer>{
    @Inject
    lateinit var em: EntityManager

    fun save(customer: Customer): Customer {
        return em.merge(customer)
    }
}